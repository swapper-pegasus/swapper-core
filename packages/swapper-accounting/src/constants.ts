import {
  Token
} from '@swapper-org/swapper-coingecko-client'

export const SYMBOL_ETH = 'eth'
export const SUPPORTED_TOKENS = [SYMBOL_ETH]
export const WEI = 1e18
export const DEFAULT_TOKEN: Token = {
  id: 'ethereum',
  name: 'Ethereum',
  symbol: 'eth'
}
