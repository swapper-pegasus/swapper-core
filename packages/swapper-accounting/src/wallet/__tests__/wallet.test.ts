/* eslint-disable @typescript-eslint/no-var-requires */
import * as Constans from '../../constants'

describe('Wallet', () => {
  beforeEach(() => jest.resetModules())
  it('getBalances', async () => {
    jest.mock('../../constants.ts', () => ({
      ...Constans,
      SUPPORTED_TOKENS: [Constans.SYMBOL_ETH, Constans.SYMBOL_BTC],
      MAP_WALLET_CONSTRUCTORS: {
        [Constans.SYMBOL_ETH]: Constans.WALLET_ETH,
        [Constans.SYMBOL_BTC]: Constans.WALLET_BTC
      }
    }))
    const { getBalances } = require('../wallet.ts')
    const result = await getBalances('')
    expect(result).toMatchSnapshot()
  })
  it('throw error if there is not wallet for token symbol', async () => {
    jest.mock('../../constants.ts', () => ({
      ...Constans,
      SUPPORTED_TOKENS: ['RRR']
    }))
    try {
      const { getBalances } = require('../wallet.ts')
      await getBalances('')
    } catch (e) {
      expect(e).toMatchSnapshot()
    }
  })
  it('missing symbol on coingecko response', async () => {
    jest.mock('../../constants.ts', () => ({
      ...Constans,
      SUPPORTED_TOKENS: [Constans.SYMBOL_ETH, Constans.SYMBOL_BTC, 'RRR'],
      MAP_WALLET_CONSTRUCTORS: {
        [Constans.SYMBOL_ETH]: Constans.WALLET_ETH,
        [Constans.SYMBOL_BTC]: Constans.WALLET_BTC,
        RRR: Constans.WALLET_BTC
      }
    }))
    try {
      const { getBalances } = require('../wallet.ts')
      await getBalances('')
    } catch (e) {
      expect(e).toMatchSnapshot()
    }
  })
})
