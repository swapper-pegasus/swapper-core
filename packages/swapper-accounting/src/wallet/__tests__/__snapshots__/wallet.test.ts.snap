// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet getBalances 1`] = `
Array [
  Object {
    "balance": Object {
      "confirmed": "0.00000",
      "unconfirmed": "0.00000",
    },
    "name": "Ethereum",
    "percentageOfPortfolio": "0",
    "priceInUsd": "1000",
    "symbol": "eth",
  },
]
`;

exports[`Wallet missing ennvironment variables 1`] = `[Error: Not env vars defined for token eth]`;

exports[`Wallet missing symbol on coingecko response 1`] = `[Error: Not converter defined for token btc]`;

exports[`Wallet throw error if there is not wallet for token symbol 1`] = `[Error: Not wallet defined for token RRR]`;
